endpoint/endpoint.go

```diff
@@ -402,49 +414,86 @@ func RemoveDuplicates(endpoints []*Endpoint) []*Endpoint {
 func (e *Endpoint) CheckEndpoint() bool {
 	switch recordType := e.RecordType; recordType {
 	case RecordTypeMX:
-		return e.Targets.ValidateMXRecord()
+		_, err := e.Targets.ParseMXRecord()
+		return err == nil
 	case RecordTypeSRV:
-		return e.Targets.ValidateSRVRecord()
+		_, err := e.Targets.ParseSRVRecord()
+		return err == nil
 	}
 	return true
 }
 
-func (t Targets) ValidateMXRecord() bool {
+func (t Targets) ParseMXRecord() ([]MXTarget, error) {
+	var mxTargets []MXTarget
+
 	for _, target := range t {
+		var mxTarget MXTarget
 		// MX records must have a preference value to indicate priority, e.g. "10 example.com"
 		// as per https://www.rfc-editor.org/rfc/rfc974.txt
 		targetParts := strings.Fields(strings.TrimSpace(target))
 		if len(targetParts) != 2 {
-			log.Debugf("Invalid MX record target: %s. MX records must have a preference value to indicate priority, e.g. '10 example.com'", target)
-			return false
+			err := fmt.Errorf("invalid MX record target: %s. MX records must have a preference value and a host, e.g. '10 example.com'", target)
+			log.Error(err)
+			return []MXTarget{}, err
 		}
-		preferenceRaw := targetParts[0]
-		_, err := strconv.ParseUint(preferenceRaw, 10, 16)
+
+		parsedPriority, err := strconv.ParseUint(targetParts[0], 10, 16)
 		if err != nil {
-			log.Debugf("Invalid SRV record target: %s. Invalid integer value in target.", target)
-			return false
+			err := fmt.Errorf("invalid integer value in target: %s", target)
+			log.Error(err)
+			return []MXTarget{}, err
 		}
+
+		mxTarget.Priority = uint16(parsedPriority)
+		mxTarget.Host = targetParts[1]
+		mxTargets = append(mxTargets, mxTarget)
 	}
-	return true
+
+	return mxTargets, nil
 }
 
-func (t Targets) ValidateSRVRecord() bool {
+func (t Targets) ParseSRVRecord() ([]SRVTarget, error) {
```

Comment:
we don't need it yet, do we?
there's only 1 caller and it doesn't need the parsed fields